caffe2/utils/CMakeLists.txt

```diff
@@ -1,47 +1,93 @@
-# ---[ GPU files
-# ------[ cuDNN
-file(GLOB_RECURSE tmp *_cudnn.cc)
-set(Caffe2_GPU_SRCS ${Caffe2_GPU_SRCS} ${tmp})
-# ------[ general GPU
-file(GLOB_RECURSE tmp *_gpu.cc)
-set(Caffe2_GPU_SRCS ${Caffe2_GPU_SRCS} ${tmp})
-# ------[ CUDA sources
-file(GLOB_RECURSE tmp *.cu)
-set(Caffe2_GPU_SRCS ${Caffe2_GPU_SRCS} ${tmp})
-# exclude test files
-file(GLOB_RECURSE tmp *_test.cc)
-exclude(Caffe2_GPU_SRCS "${Caffe2_GPU_SRCS}" ${tmp})
-
-# ---[ CPU files.
-file(GLOB_RECURSE tmp *.cc)
-set(Caffe2_CPU_SRCS ${Caffe2_CPU_SRCS} ${tmp})
-# exclude test files and gpu files
-file(GLOB_RECURSE tmp *_test.cc)
-exclude(Caffe2_CPU_SRCS "${Caffe2_CPU_SRCS}" ${tmp})
-exclude(Caffe2_CPU_SRCS "${Caffe2_CPU_SRCS}" ${Caffe2_GPU_SRCS})
+set(OLD_LIB_SOURCES_CPU
+        utils/proto_utils.cc
+        utils/murmur_hash3.cc
+        utils/smart_tensor_printer.cc
+        utils/signal_handler.cc
+        utils/string_utils.cc
+        utils/threadpool/ThreadPool.cc
+        utils/cpuid.cc
+        utils/bench_utils.cc
+        utils/math_cpu.cc
+        utils/math_utils.cc
+        )
 
 # ---[ threadpool/pthreadpool* is a local modification of the NNPACK
 # pthreadpool with a very similar interface. Neither NNPACK, nor this
 # thread pool supports Windows.
-if (MSVC)
-  file(GLOB_RECURSE tmp pthreadpool*)
-  exclude(Caffe2_CPU_SRCS "${Caffe2_CPU_SRCS}" ${tmp})
+if (NOT MSVC)
+  list(APPEND OLD_LIB_SOURCES_CPU
+          utils/threadpool/pthreadpool.cc
+          utils/threadpool/pthreadpool_impl.cc
+          )
 endif()
 
-# ---[ GPU test files
-file(GLOB_RECURSE tmp *_gpu_test.cc)
-set(Caffe2_GPU_TEST_SRCS ${Caffe2_GPU_TEST_SRCS} ${tmp})
+set(OLD_LIB_SOURCES_GPU
+        utils/math_gpu.cu
+        )
 
-# ---[ CPU test files
-file(GLOB_RECURSE tmp *_test.cc)
-set(Caffe2_CPU_TEST_SRCS ${Caffe2_CPU_TEST_SRCS} ${tmp})
-exclude(Caffe2_CPU_TEST_SRCS "${Caffe2_CPU_TEST_SRCS}" ${Caffe2_GPU_TEST_SRCS})
+set(OLD_TEST_SOURCES_CPU
+        utils/fixed_divisor_test.cc
+        utils/math_test.cc
+        utils/fatal_signal_asan_no_sig_test.cc
+        utils/simple_queue_test.cc
+        utils/proto_utils_test.cc
+        utils/cpuid_test.cc
+        utils/smart_tensor_printer_test.cc
+        utils/cast_test.cc
+        )
 
-# ---[ Send the lists to the parent scope.
-set(Caffe2_CPU_SRCS ${Caffe2_CPU_SRCS} PARENT_SCOPE)
-set(Caffe2_GPU_SRCS ${Caffe2_GPU_SRCS} PARENT_SCOPE)
-set(Caffe2_CPU_TEST_SRCS ${Caffe2_CPU_TEST_SRCS} PARENT_SCOPE)
-set(Caffe2_GPU_TEST_SRCS ${Caffe2_GPU_TEST_SRCS} PARENT_SCOPE)
+set(OLD_TEST_SOURCES_GPU
+        utils/math_gpu_test.cc
+        )
 
-# ---[ Printing stack traces requires dladdr
+# TODO Remove the OLD_xxx variables above and add them to the variables for the local library target below instead
```

Comment:
Also, now that I think about it, can we not use `OLD_xxx`? I'm not even sure if we'll go down this modularity path, so we should really optimize for where we are right now. Please just call this something else and/or add it to `Caffe2_*` direct.